src/state_machine/coordinator/mod.rs

```diff
@@ -972,4 +973,71 @@ pub mod test {
 
         assert_eq!(signers, loaded_signers);
     }
+
+    pub fn gen_nonces<Coordinator: CoordinatorTrait, SignerType: SignerTrait>(
+        num_signers: u32,
+        keys_per_signer: u32,
+    ) {
+        let mut rng = OsRng;
+
+        let (mut coordinators, mut signers) =
+            run_dkg::<Coordinator, SignerType>(num_signers, keys_per_signer);
+
+        let msg = "It was many and many a year ago, in a kingdom by the sea"
+            .as_bytes()
+            .to_vec();
+
+        let signature_type = SignatureType::Frost;
+
+        // Start a signing round
+        let message = coordinators
+            .first_mut()
+            .unwrap()
+            .start_signing_round(&msg, signature_type)
+            .unwrap();
+        assert_eq!(
+            coordinators.first_mut().unwrap().get_state(),
+            State::NonceGather(signature_type)
+        );
+
+        // Send the message to all signers and gather responses by sharing with all other signers and coordinator
+        let (outbound_messages, operation_results) =
+            feedback_messages(&mut coordinators, &mut signers, &[message]);
+        assert!(operation_results.is_empty());
+        assert_eq!(
+            coordinators.first_mut().unwrap().get_state(),
+            State::SigShareGather(signature_type)
+        );
+
+        assert_eq!(outbound_messages.len(), 1);
+        match &outbound_messages[0].msg {
+            Message::SignatureShareRequest(_) => {}
+            _ => {
+                panic!("Expected SignatureShareRequest message");
+            }
+        }
+
+        let messages1 = signers[0]
+            .process(&outbound_messages[0].msg, &mut rng)
+            .unwrap();
+
+        let messages2 = signers[0]
+            .process(&outbound_messages[0].msg, &mut rng)
+            .unwrap();
+
+        for (message1, message2) in messages1.into_iter().zip(messages2) {
+            let share1 = if let Message::SignatureShareResponse(response) = message1 {
+                response.signature_shares[0].clone()
+            } else {
+                panic!("Message should have been SignatureShareResponse");
+            };
+            let share2 = if let Message::SignatureShareResponse(response) = message2 {
+                response.signature_shares[0].clone()
+            } else {
+                panic!("Message should have been SignatureShareResponse");
+            };
+
+            assert!(share1.z_i != share2.z_i);
```

Comment:
I think the standard thing to do here is to use the `assert_ne!` macro for this.